chroot_setup

```diff
@@ -0,0 +1,428 @@
+  if not CHROOT_DEBUG:
+    os.makedirs(f"{args.chroot_directory}{dir}", exist_ok=True)
+
+# :ln
+# Create an absolute symlink. We don't have to go inside the chroot, since the
+# links are resolved as they are used. The target doesn't even have to exist,
+# Python won't check and therefore won't throw an error.
+def chroot_ln (line: str):
+  ln_command = line.split()
+  assert len(ln_command) == 2, ":ln action expects 2 arguments!"
+  target = ln_command[0]
+  link_name = ln_command[1]
+  print(f"os.symlink({target}, {args.chroot_directory}{link_name})")
+  if not CHROOT_DEBUG:
+    os.symlink(target, f"{args.chroot_directory}{link_name}")
+
+# :ln_rel
+# Create a relative symlink in the chroot that mirrors a relative symlink that
+# exists outside of the chroot.
+def chroot_ln_rel (line: str):
+  ln_command = line.split()
+  assert len(ln_command) == 1, ":ln_fix_rel action expects 1 argument!"
+  directory = os.path.dirname(line)
+  basename = os.path.basename(line)
+  target = os.readlink(line)
+  here = os.getcwd()
+  print(f"os.chdir({args.chroot_directory}{directory})")
+  print(f"os.symlink({target}, {basename})")
+  if not CHROOT_DEBUG:
+    os.chdir(f"{args.chroot_directory}{mydir}")
+    os.symlink(target, basename)
+    os.chdir(here)
+
+# :ln
+# Create a relative symlink inside of the chroot, specifying where the link
+# should be pointed to. This means you can change a relative symlink you pass
+# in, or simply create a relative symlink where one didn't exist before
+def chroot_ln_fix_rel (line: str):
+  ln_command = line.split()
+  assert len(ln_command) == 2, ":ln_fix_rel action expects 2 arguments!"
+  target, link_name = ln_command
+  directory = os.path.dirname(link_name)
+  basename = os.path.basename(link_name)
+  here = os.getcwd()
+  print(f"os.chdir({args.chroot_directory}{directory})")
+  print(f"os.symlink{target}, {basename})")
+  print(f"os.chdir({here})")
+  if not CHROOT_DEBUG:
+    os.chdir(f"{args.chroot_directory}{mydir}")
+    os.symlink(target, basename)
+    os.chdir(here)
+
+# :mknod
+# Create a node with a given name, type, and major and minor number
+def chroot_mknod (line: str):
+  # We are passed a mknod terminal commmand. Need to parse it to make it work
+  # with os.mknod.
+  mknod_command = line.split()
+  assert len(mknod_command) == 4, "mknod command expects 4 arguments!"
+  node: str  = mknod_command[0]
+
+  # we only support creating character special devices
+  if mknod_command[1] != 'c':
+    print("Unrecognized mknod type; Valid type is 'c'")
+    sys.exit(-6)
+
+  mode = 0o644 | stat.S_IFCHR
+
+  # device major and minor numbers are created through os.makedev
+  major = mknod_command[2]
+  minor = mknod_command[3]
+  device = os.makedev(int(major), int(minor))
+
+  print(f"os.remove({args.chroot_directory}{node})")
+  print(f"os.mknod({args.chroot_directory}{node}, {mode}, {device})")
+  if not CHROOT_DEBUG:
+    os.remove(f"{args.chroot_directory}{node}")
```

Comment:
This remove should be wrapped in a try catch. It should only be removed when the it already exists. On initial chroot setup it does not exist and this line raises a `FileNotFoundError`.

```suggestion
    try:
      os.remove(f"{args.chroot_directory}{node}")
    except FileNotFoundError:
      pass
```

Reply:
Fixed up in e1164d1